service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/SendNegativeAcknowledgementTaskDefinition.java

```diff
@@ -0,0 +1,22 @@
+package uk.nhs.adaptors.gp2gp.ehr;
+
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.experimental.SuperBuilder;
+import lombok.extern.jackson.Jacksonized;
+import uk.nhs.adaptors.gp2gp.common.task.TaskType;
+
+import static uk.nhs.adaptors.gp2gp.common.task.TaskType.SEND_NEGATIVE_ACKNOWLEDGEMENT;
+
+@Jacksonized
+@SuperBuilder
+@Getter
+@EqualsAndHashCode(callSuper = true)
+public class SendNegativeAcknowledgementTaskDefinition extends SendAcknowledgementTaskDefinition {
+    private final String reasonCode;
+    private final String detail;
```

Comment:
As these two fields exist in the base class, should we get rid of them?

Reply:
Now I see that the base ones are Optionals. In such case, maybe we should make this definition independent from `SendAcknowledgementTaskDefinition` in terms of inheritance and just define its own set of fields in it? There's going to be very few of them, anyway.